internals-js/src/operations.ts

```diff
@@ -1071,8 +1082,8 @@ export class SelectionSet extends Freezable<SelectionSet> {
     }
   }
 
-  optimize(fragments?: NamedFragments): SelectionSet {
-    if (!fragments || fragments.isEmpty()) {
+  optimize(fragments?: NamedFragments, options: OptimizeOptions = {}): SelectionSet {
+    if (!fragments || (!options.autoFragmetize && fragments.isEmpty())) {
```

Comment:
Same remark that on `Operation` regarding `!fragments`. In that case however, it's probably stranger to create a new `NamedFragments` objects. But we actually never pass `undefined` to this particular method (we do for `Operation.optimize` however), so I might suggest just making `fragments` mandatory here.